jwst/datamodels/library.py

```diff
@@ -212,6 +214,43 @@ def _assign_member_to_model(self, model, member):
         if "asn_pool" in self.asn.keys():  # do not clobber existing values
             model.meta.asn.pool_name = self.asn["asn_pool"]
 
+    def get_crds_parameters(self):
+        """
+        Override base method so lazy-loading of metadata can be used to get CRDS parameters.
+
+        Get the "crds_parameters" from either:
+            - the first "science" member (based on exptype)
+            - the first model (if no "science" member is found)
+
+        If no "science" members are found in the library a ``UserWarning``
+        will be issued.
+
+        Returns
+        -------
+        crds_parameters : dict
+            The result of ``get_crds_parameters`` called on the selected
+            model.
+        """
+        with self:
+            science_index = None
+            for i, member in enumerate(self._members):
+                if member["exptype"].lower() == "science":
+                    science_index = i
+                    break
+            if science_index is None:
+                warnings.warn(
+                    "get_crds_parameters failed to find any science members. "
+                    "The first model was used to determine the parameters",
+                    UserWarning,
+                    stacklevel=2,
+                )
+                science_index = 0
+
+            # Retrieve the filename
+            filename = Path(self._asn_dir) / member["expname"]
+            parameters = read_metadata(filename, flatten=True)
```

Comment:
Does this work for a `ModelLibrary` that:
- is "on_disk" and models have been modified (where the new models are no longer in `_asn_dir`?
- is "in_memory" where a model has been modified?